rust/noosphere/src/ffi/authority.rs

```diff
@@ -0,0 +1,125 @@
+use std::ffi::c_void;
+
+use cid::Cid;
+use noosphere_core::{
+    authority::Authorization,
+    data::{Did, Mnemonic},
+};
+use noosphere_sphere::{HasMutableSphereContext, SphereAuthorityEscalate, SphereAuthorityWrite};
+use safer_ffi::prelude::*;
+
+use crate::{
+    error::NoosphereError,
+    ffi::{NsError, NsNoosphere, NsSphere},
+};
+
+#[ffi_export]
+/// @memberof ns_sphere_t
+///
+/// Authorize another key to manipulate this sphere, given a display name for
+/// the authorization and the DID of the key to be authorized.
+///
+/// The callback arguments are (in order):
+///
+///  1. The context argument provided in the original call to
+///     ns_sphere_authority_authorize
+///  2. An owned pointer to an ns_error_t if there was an error, otherwise NULL
+///  3. An owned pointer to constant string (the authorization CID) if the call
+///     was successful, otherwise NULL
+pub fn ns_sphere_authority_authorize(
```

Comment:
Requiring root authority explains some of my earlier questions I think, but some Q's:

* Why require root access? Wouldn't the primary/original author be able to delegate authority to another sphere, in terms of ucan?
* Certainly increases the necessity of storing the mnemonic on generation if primary author is not granted this permissions, and ultimately increases the necessity of good UX around this

Reply:
Root is currently only required for revocation; authorization can be done by anyone.

This is a shortcut. Revocation by a delegated key should be possible, but the UX is more complicated. For example: a delegatee cannot revoke authority from its delegator, and this nuance would have to be reflected in the UI on each device.

As we shore up the UX for delegation and revocation, we can begin to explore patterns that do not call for authority escalation.